grails-app/services/de/iteratec/osm/measurement/environment/wptserver/LocationPersisterService.groovy

```diff
@@ -99,4 +103,36 @@ class LocationPersisterService implements iLocationListener {
             }
         }
     }
+
+    /**
+     * Parses the location label to determine the possible device type
+     * @param label location label for wptServer
+     */
+    DeviceType parseDeviceType(String label) {
+        switch (label) {
+            case ~/(?i).*(-Win|IE\\s*[1-9]*|firefox|nuc).*/ :
+                return DeviceType.DESKTOP
+            case ~/(?i).*(Pad|Tab|Note|Xoom|Book|Tablet).*/ :
+                return DeviceType.TABLET
+            case ~/(?i)(?!(.*(Pad|Tab|Note|Xoom|Book|Tablet).*)).*(Samsung|Moto|Sony|Nexus|Huawei|Nokia|Alcatel|LG|OnePlus|HTC|Phone).*/ :
+                return DeviceType.SMARTPHONE
+            default: return DeviceType.UNDEFINED
+        }
+    }
+
+    /**
+     * Parses the location label to determine the possible operating system
+     * @param label location label for wptServer
+     */
+    OperatingSystem parseOperatingSystem(String label) {
```

Comment:
Suggestion: move this to `OperatingSystem` as a static method `guessOperatingSystem`, so others can easily profit from this logic.